nimble/host/mesh/src/prov.c

```diff
@@ -1195,6 +1221,12 @@ static void prov_retransmit(struct ble_npl_event *work)
 	}
 }
 
+static void prot_tmo_link_close(struct ble_npl_event *ev)
+{
+    close_link(PROV_ERR_UNEXP_ERR, CLOSE_REASON_FAILED);
+    reset_link();
```

Comment:
`reset_link()` is a PB-ADV specific internal API, so calling it from this function doesn't seem correct. Also, passing a non-zero first parameter to `close_link()` doesn't seem right since that would cause a second (duplicate) failure message to be sent. It would also be in violation of the spec text " In the case of protocol timeout, the device shall not send a Provisioning Failed PDU.".

Reply:
Actually it seems Zephyr and nimble are out of sync with each other regarding the `reset_link()` function's purpose. There's `reset_link()` and `reset_state()` in Zephyr. I'm renaming the former to `reset_adv_link()`.

Reply:
Is it better to use bearer_ctl_send(LINK_CLOSE, NULL, 0) here? 
And after that clear the LINK_INVALID bit with: atomic_clear_bit(link.flags, LINK_INVALID)?

Reply:
That's essentially what I've done (in my yet to be pushed backporting work). The link_close() function went away completely since the timeout was the only remaining user of it. Note that the link close message has a reason parameter and you should use CLOSE_REASON_TIMEOUT (0x02) for it.

Reply:
Also, I've put the clear_bit(LINK_INVALID) inside the prov_send_fail_msg() function since it was always cleared right after it anyway.

Reply:
So you've put clear_bit(LINK_INVALID) just after bearer_ctl_send() and another in prov_send_fail_msg() right? Also, I think that CLOSE_REASON_TIMEOUT has 0x01 is that correct?

Reply:
Sorry, I meant set_bit(LINK_INVALID) - only place I found was needed was at the end of prof_send_fail_msg(). And yes, you're correct about the timeout error code.